src/glum/_glm.py

```diff
@@ -1552,6 +1625,53 @@ def score(
         dev_null = family.deviance(y, y_mean, sample_weight=sample_weight)
         return 1.0 - dev / dev_null
 
+    def _compute_information_criteria(self):
+        """
+        Computes and stores the model's degrees of freedom, and the 'aic',
+        'aicc' and 'bic' information criterion for the model.
+
+        References
+        ----------
+        [1] Burnham KP, Anderson KR (2002). Model Selection and Multimodel
+        Inference; Springer New York.
+        """
+
+        # we require that the log_likelihood be definied
+        if not isinstance(
+            self.family_instance,
+            (
+                BinomialDistribution,
+                GammaDistribution,
+                NormalDistribution,
+                PoissonDistribution,
+            ),
+        ):
```

Comment:
```suggestion
        if not isinstance(
            self.family_instance, (BinomialDistribution, TweedieDistribution)
        ):
```

Reply:
Hey Luca. So these information criteria are typically defined as in sample measures only. So if you do actually have out of sample data (e.g., validation or test) then it is much better to use standard cross-validation or just optimisation on the validation set using held out log-likelihood. My suggestion is to force this best practice by not giving `X` and `y` as arguments -- i.e., if someone finds they are computing these on the validation set something has gone wrong.

Reply:
I can make the change above but there are two things to note:

1. The `log_likelihood` throws a `NotImplementedError` for the `InverseGaussianDistribution` (TweedieDistribution(p=3)). So we'd need a try catch here to catch that exception.
2. The `log_likelihood` method appears to hang for values of p \in (1,2). I think this has to do with the series approximation that is mentioned in the docs. This might be a bug that I can create an issue for but it will cause this method to hang indefinitely for now. This can be managed by checking for self.power in [0,1,2] but this is going to be a more verbose piece of code. What do you think?

Reply:
> My suggestion is to force this best practice by not giving `X` and `y` as arguments

The function is private anyway, so I don't know that the benefits outweigh the considerably more complicated implementation.

> we'd need a try catch here to catch that exception.

You can also check the `power` attribute of Tweedie distributions and skip the information criteria if it isn't in the [0, 2] interval.

> The `log_likelihood` method appears to hang for values of p in (1,2). 

Strange. We have tests ([here](https://github.com/Quantco/glum/blob/b072bd652a8c24295acadde63c9253f262bc3d71/tests/glm/test_distribution.py#L400) and [here](https://github.com/Quantco/glum/blob/b072bd652a8c24295acadde63c9253f262bc3d71/tests/glm/test_distribution.py#L468)), for a variety of parametrizations, and we don't run into deadlocks. Perhaps you're running into parallelization issues? Can you run those tests locally? Seems worth investigating.

Reply:
> The function is private anyway, so I don't know that the benefits outweigh the considerably more complicated implementation.

Yep thats a good point. I'll update this. 

> You can also check the power attribute of Tweedie distributions and skip the information criteria if it isn't in the [0, 2] interval.

I'll make this change as well.

> Strange. We have tests ([here](https://github.com/Quantco/glum/blob/b072bd652a8c24295acadde63c9253f262bc3d71/tests/glm/test_distribution.py#L400) and [here](https://github.com/Quantco/glum/blob/b072bd652a8c24295acadde63c9253f262bc3d71/tests/glm/test_distribution.py#L468)), for a variety of parametrizations, and we don't run into deadlocks. Perhaps you're running into parallelization issues? Can you run those tests locally? Seems worth investigating.

The tests work as is. However, if I change the `method` argument of the dispersion calculation to default ( method="pearson") then the tests hangs ([line 406](https://github.com/Quantco/glum/blob/b072bd652a8c24295acadde63c9253f262bc3d71/tests/glm/test_distribution.py#L406)). I think it has to do with the dispersion being really small in the tests (`8.213159509438382e-23`). It might be some kind of underflow or something. I am still not too sure why it hangs though.